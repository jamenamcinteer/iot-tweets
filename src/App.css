body {
  background-color: #0A111C;
}

.grid {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-column-gap: 150px;
  width: 100%;
}

.grid__chart {
  grid-column: 2;
}
.grid__chartFrame {
  width: 400px;
  margin: 0 auto;
}

.grid__tweets {
  grid-column: 3;
}
.grid__tweetsContainer {
  max-width: 400px;
  display: flex;
  align-items: center;
  margin: 0 auto;
  height: 100%;
}

.grid__tweetsContainer img {
    -webkit-mask-image: url('data:image/svg+xml;utf8,<svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg"><circle cx="50" cy="50" r="50"/></svg>');
    mask-image: url('data:image/svg+xml;utf8,<svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg"><circle cx="50" cy="50" r="50"/></svg>');
    mask-mode: alpha;
    -webkit-mask-repeat: no-repeat;
    mask-repeat: no-repeat;
    -webkit-mask-size: contain;
    mask-size: contain;
    -webkit-mask-position: center;
    mask-position: center;
    border-radius: 9999px;
}

.tweet__date {
  white-space: nowrap;
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}
