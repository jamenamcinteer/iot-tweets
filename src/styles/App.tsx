import { IPropsTheme } from "../interfaces/interfaces"
import styled from "styled-components";

export const AppContainer = styled.main`
  text-align: center;
`

export const AppBody = styled.div`
  background-color: rgba(0, 0, 0, 0.7);
  color: ${(props: IPropsTheme) => props.theme.baseFontColor};
  min-height: calc(100vh - 23rem);
  padding-bottom: 5rem;
`

export const AppHeader = styled.div`
  background-color: rgba(0, 0, 0, 0.7);
  margin: 0;
  padding-bottom: 4rem;

  @media (min-width: 80rem) {
    padding-bottom: 8rem;
  }
`

export const AppHeaderText = styled.h1`
  font-size: 3rem;
  margin: 0;
  padding: 3rem 0 1rem 0;
`

export const AppHeaderSubtext = styled.h2`
  font-size: 1.4rem;
  font-weight: normal;
  margin: 0;
`