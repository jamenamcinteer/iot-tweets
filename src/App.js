import React from 'react';
import MainLayout from "./components/MainLayout";
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <MainLayout />
      </header>
    </div>
  );
}

export default App;
