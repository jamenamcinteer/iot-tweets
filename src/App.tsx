import React, { useState, useEffect } from 'react'
import { ThemeProvider } from "styled-components"
import MainLayout from "./components/MainLayout"
import theme from "./theme"
import { ITweets } from "./interfaces/interfaces"
import { GlobalStyles } from "./styles/GlobalStyles"
import { AppContainer, AppBody, AppHeader, AppHeaderText, AppHeaderSubtext } from "./styles/App"


function App() {
  const [twitterData, setTwitterData] = useState<Array<ITweets> | []>([])
  const [apiError, setApiError] = useState<boolean>(false)

  // Fetch the twitter data from our API
  useEffect(() => {
    async function fetchData(): Promise<void> {
      try {
        const response: Response = await fetch('/tweets')
        if(response.status !== 200) {
          setApiError(true)
        }
        else {
          const json: Array<ITweets> = await response.json()
          setTwitterData(json)
        }
      } catch (error) {
        setApiError(true)
      }
    }
    fetchData()
  }, [])

  return (
    <ThemeProvider theme={theme}>
      <AppContainer>
        <GlobalStyles />
        <AppHeader>
          <AppHeaderText>#IoT</AppHeaderText>
          <AppHeaderSubtext>Top 20 Words in 100 Recent Tweets</AppHeaderSubtext>
        </AppHeader>
        <AppBody>
          {
            !apiError
            ? <MainLayout twitterData={twitterData} />
            : <div>Sorry, there was an issue with fetching the Twitter data. Please try again later.</div>
          }
        </AppBody>
      </AppContainer>
    </ThemeProvider>
    
  )
}

export default App
