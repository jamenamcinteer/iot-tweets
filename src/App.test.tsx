import React from 'react';
import { cleanup, render, wait } from "@testing-library/react";
import "jest-styled-components";
import { mockSuccessResponseTwitter } from "./__mocks__/mockTwitterApiResponse"
import App from './App';

afterEach(cleanup);

it('renders without crashing', async () => {
  render(<App />)
})

it('calls the API when rendered', async () => {
  const mockJsonPromise = Promise.resolve(mockSuccessResponseTwitter);
  const mockFetchPromise = Promise.resolve({
    json: () => mockJsonPromise
  });
  jest.spyOn(global, "fetch").mockImplementation(() => mockFetchPromise);

  render(<App />)
  await wait(() => expect(global.fetch).toHaveBeenCalledTimes(1))
})

it('handles an error from the API call', async () => {
  const mockFetchPromise = Promise.reject({
    error: 'rejected'
  });
  jest.spyOn(global, "fetch").mockImplementation(() => mockFetchPromise);

  render(<App />)
  
  await wait(() => expect(global.fetch).toHaveBeenCalledTimes(2))
})