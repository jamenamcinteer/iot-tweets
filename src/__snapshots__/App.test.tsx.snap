// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles an error from the API call 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
}

.c4 {
  background-color: rgba(0,0,0,0.7);
  color: #FFFFFF;
  min-height: calc(100vh - 23rem);
  padding-bottom: 5rem;
}

.c1 {
  background-color: rgba(0,0,0,0.7);
  margin: 0;
  padding: 0 2rem;
  padding-bottom: 4rem;
}

.c2 {
  font-size: 3rem;
  margin: 0;
  padding: 3rem 0 1rem 0;
}

.c3 {
  font-size: 1.4rem;
  font-weight: normal;
  margin: 0;
}

@media (min-width:40rem) {
  .c1 {
    padding: 0;
    padding-bottom: 4rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    padding-bottom: 8rem;
  }
}

<main
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        #IoT
      </h1>
      <h2
        class="c3"
      >
        Top 20 Words in 100 Recent Tweets
      </h2>
    </div>
    <div
      class="c4"
    >
      <div>
        Sorry, there was an issue with fetching the Twitter data. Please try again later.
      </div>
    </div>
  </main>
</DocumentFragment>
`;
